R/serializer.R

```diff
@@ -231,6 +231,26 @@ serializer_unboxed_json <- function(auto_unbox = TRUE, ..., type = "application/
   serializer_json(auto_unbox = auto_unbox, ..., type = type)
 }
 
+#' @describeIn serializers GeoJSON serializer. See also [geojsonsf::sf_geojson()] and [[geojsonsf::sfc_geojson()]].
+#' @inheritParams geojsonsf::sf_geojson
+#' @inheritParams geojsonsf::sfc_geojson
+#' @export
+serializer_geojson <- function(..., type = "application/geo+json") {
+  serializer_content_type(type, function(val) {
+    if (any(class(val) == "sfc")) {
+
+      geojsonsf::sfc_geojson(val, ...)
+
+    } else if (any(class(val) == "sf")) {
+
+      geojsonsf::sf_geojson(val, ...)
+
+    } else {
+      stop('Object must be of class "sf" or "sfc"', call. = FALSE)
+    }
```

Comment:
Two changes:
* Function change: Let's use `inherits(val, classVal)` instead of `any(class(val) == classVal)`
* Cosmetic: I'm a fan of early returns, rather than `else if` of death.

```suggestion
    if (inherits(val, "sfc")) return(geojsonsf::sfc_geojson(val, ...))
    if (inherits(val, "sf"))   return(geojsonsf::sf_geojson(val, ...))
    stop('Object must be of class "sf" or "sfc"', call. = FALSE)
```